web/oidc/oidc.go

```diff
@@ -94,6 +94,19 @@ func Login(c echo.Context) error {
 		return renderError(c, inst, http.StatusBadRequest, "Sorry, the cozy was not found.")
 	}
 
+	if inst.HasAuthMode(instance.TwoFactorMail) {
+		twoFactorToken, err := lifecycle.SendTwoFactorPasscode(inst)
+		if err != nil {
+			return err
+		}
+		v := url.Values{}
+		v.Add("two_factor_token", string(twoFactorToken))
+		v.Add("trusted_device_checkbox", "false")
+		// We can not cleanly check the trusted_device option for external
+		// login. Therefore, we do not provide the checkbox
```

Comment:
The trusted device checkbox allows a user to trust a device and skip the 2FA authentication. 

This is done by storing a token in the `localStorage` of the user's browser, and reading/appending it during the passphrase login form.

In the case of an external login (like OIDC for example), we cannot easily get this token because the first rendered stack page will be the 2FA page (if activated).

Even if we decide to store & check the token on the 2FA page, the user will anyway get an email with the 2FA code as we can only sent it during the OIDC Login function for security reasons.